File: Selection Sort.c
#include<stdio.h>

int n=6;

void selection(int a[],int n)           //function for sorting
{
    int i , j , min , swap ;
	
	for(i=0;i<n-1;i++)
	{
		min=i;               ////minimum element in unsorted array
		
		for(j=i+1;j<n;j++)
		{
			if(a[j]<a[min])
			{
				min=j;
			}
		}
		
		if(min!=i)
		{
		    swap=a[i];
		    a[i]=a[min];
		    a[min]=swap;                   // Swap the minimum element with the first element  
			
		}
	}
}
void display(int a[], int n)             // function to print the array 
{  
    int i;  
    for (i = 0; i < n; i++)  
        printf("%d ", a[i]);  
}  

int main()
{
	
    int arr[6]={7,4,10,8,3,1};
  
    printf("Before sorting array elements are - \n");  
    display(arr, n);  
    selection(arr, n);  
    printf("\nAfter sorting array elements are - \n");    
    display(arr, n); 
}

File: selection sort.java
import java.util.*;

public class Main
{   // function for sorting

    public static void Sort(int arr[])
    {
        int  n = arr.length;
        
        for(int i=0 ; i<n-1;i++)
        {
            int smallest=i;
            
            for(int j=i+1; j<n;j++)
            {
                if(arr[smallest]>arr[j])
                {
                    smallest=j;
                }
            }
            int temp=0;
        
        //swap
        
        temp=arr[smallest];
        arr[smallest]=arr[i];
        arr[i]=temp;
        }
        
        
    }
    
    // function for display
    
    public static void display(int arr[])
    {
        int n = arr.length;
        
        for(int i=0;i<n;i++)
        {
            System.out.print(arr[i]+" ");  
        }
        
        System.out.println();
    }
    
    
    
    public static void main(String[] args) {
        
        
        
        
        Scanner scan = new Scanner(System.in);
        
		System.out.println("Enter the length of the array: ");
		int n= scan.nextInt();
		
		int arr[]= new int[n];
		
		System.out.println("Enter the array elements: ");
		
		for(int i=0;i<n;i++)
		{
		    arr[i]= scan.nextInt();
		}
		
		System.out.println("Elements before Sorting: ");
		display(arr);
		Sort(arr);
		System.out.println("Elements after Sorting: ");
		display(arr);

		
	}
    
    
}
